nexus/src/app/sagas/snapshot_create.rs

```diff
@@ -142,10 +155,32 @@ declare_saga_actions! {
         + ssc_create_snapshot_record
         - ssc_create_snapshot_record_undo
     }
-    SEND_SNAPSHOT_REQUEST -> "snapshot_request" {
-        + ssc_send_snapshot_request
-        - ssc_send_snapshot_request_undo
+    SEND_SNAPSHOT_REQUEST_TO_SLED_AGENT -> "snapshot_request_to_sled_agent" {
+        + ssc_send_snapshot_request_to_sled_agent
+        - ssc_send_snapshot_request_to_sled_agent_undo
+    }
+    ATTACH_DISK_TO_PANTRY -> "attach_disk_to_pantry" {
+        + ssc_attach_disk_to_pantry
+        - ssc_attach_disk_to_pantry_undo
+    }
+    GET_PANTRY_ADDRESS -> "pantry_address" {
+        + ssc_get_pantry_address
+    }
```

Comment:
I don't think we should embed the IP addresses of services in a saga - this can make it impossible to perform undo operations if a service is rebalanced.

For example:

- Action 1: Get pantry address
- Action 2: Send operation to pantry, using that address. Allocates a "Foo" object.
- Suppose service rebalancing happens, and the pantry moves to a new sled
- Action 3: Send another operation to pantry, using the old address. This fails.
- Undo Action 2: Cannot undo the "Foo" object allocation, the old pantry address no longer exists, and never will.

IMO we should be uncovering the pantry address as we use it.

Reply:
Unfortunately, the Pantry is not a stateless service that can be rebalanced like this. It will hold an activated Volume in memory and all requests referencing that Volume will need to be sent to the same instance of the service, hence the `GET_PANTRY_ADDRESS`.

If the Pantry crashes then it will lose all that state. If it restarts, further requests referencing that Volume will return errors, and sending those requests to other Pantrys will also return errors.